test/integration/translation_failures_test.go

```diff
@@ -0,0 +1,130 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane"
+	kongv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
+	"github.com/kong/kubernetes-ingress-controller/v2/pkg/clientset"
+)
+
+// TestTranslationFailures ensures that proper warning Kubernetes events are recorded in case of translation failures
+// encountered.
+func TestTranslationFailures(t *testing.T) {
+	testCases := []struct {
+		name string
+		// translationFailureTrigger should create objects that trigger translation failure and return the objects
+		// that we expect translation failure warning events to be created for.
+		translationFailureTrigger func(ns string) []client.Object
+	}{
+		{
+			name: "invalid CA secret",
+			translationFailureTrigger: func(ns string) []client.Object {
+				createdSecret, err := env.Cluster().Client().CoreV1().Secrets(ns).Create(ctx, invalidCASecret(ns), metav1.CreateOptions{})
+				require.NoError(t, err)
+
+				return []client.Object{createdSecret}
+			},
+		},
+		{
+			name: "invalid CA secret referred by a plugin",
+			translationFailureTrigger: func(ns string) []client.Object {
+				createdSecret, err := env.Cluster().Client().CoreV1().Secrets(ns).Create(ctx, invalidCASecret(ns), metav1.CreateOptions{})
+				require.NoError(t, err)
+
+				c, err := clientset.NewForConfig(env.Cluster().Config())
+				require.NoError(t, err)
+				createdPlugin, err := c.ConfigurationV1().KongPlugins(ns).Create(ctx, pluginUsingInvalidCACert(ns), metav1.CreateOptions{})
+				require.NoError(t, err)
+
+				// expect events for both: a faulty secret and a plugin referring it
+				return []client.Object{createdSecret, createdPlugin}
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
Is this necessary? How long do these testcases run so that we have to use `.Parallel()`?

My suggestion is to drop this unless there's a substantial runtime gain (worth the garbled test output).

Reply:
These are integration tests so I'd say it's preferable to run them in parallel, especially since more is to come here. I've just measured their duration and got the following results: 
```
TestTranslationFailures/invalid_CA_secret_referred_by_a_plugin
    translation_failures_test.go:67: test case took: 8251ms
TestTranslationFailures/invalid_CA_secret
    translation_failures_test.go:67: test case took: 8211ms
```

Reply:
I missed that. Good point!